x-pack/plugins/actions/server/builtin_action_types/lib/get_proxy_agents.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Agent } from 'http';
+import HttpProxyAgent from 'http-proxy-agent';
+import { HttpsProxyAgent } from 'https-proxy-agent';
+import { Logger } from '../../../../../../src/core/server';
+import { ProxySettings } from '../../types';
+
+interface GetProxyAgentsResponse {
+  httpAgent: Agent | undefined;
+  httpsAgent: Agent | undefined;
+}
+
+export function getProxyAgents(
+  proxySettings: ProxySettings | undefined | null,
+  logger: Logger
+): GetProxyAgentsResponse {
+  const undefinedResponse = {
+    httpAgent: undefined,
+    httpsAgent: undefined,
+  };
```

Comment:
it could, but then it becomes a global.  Given the way it's used currently (returned fields are only accessed, not set), that would be fine, but we could conceivably break that usage in the future, and somehow have the global set to something else.  Feels safer to leave it as it is ...